.github/workflows/profiler-pipeline.yml

```diff
@@ -247,55 +245,18 @@ jobs:
       - name: Clone dd-trace-dotnet repository
         uses: actions/checkout@v3
 
-      - name: Build Native Profiler Engine
-        run: |
-          CXX=clang++ CC=clang cmake -S . -B __build -DRUN_ASAN=ON
-          cmake --build __build --config Release --target all-profiler
-
-      - name: Restore sample applications
-        run: dotnet restore profiler/src/Demos/Datadog.Demos.sln /p:Configuration="Release" /p:CheckEolTargetFramework=false
-
-      - name: Build sample applications
-        run: dotnet build -c Release profiler/src/Demos/Datadog.Demos.sln -p:Platform=x64 -p:CheckEolTargetFramework=false
-
-      - name: Run unit tests
-        run: |
-          sudo mkdir -p /var/log/datadog/dotnet
-          sudo chmod a+rwx /var/log/datadog/dotnet
-          ASAN_OPTIONS="detect_leaks=0" ./profiler/_build/bin/Datadog.Profiler.Native.Tests/Datadog.Profiler.Native.Tests
-
-      - name: Prepare environment to run the application
-        run: |
-          #
-          # set profiler deployment and test output dir folder path for the test
-          #
-          tests_output_dir="$(realpath .)/test_output"
-          echo "DD_TESTING_OUPUT_DIR=$tests_output_dir" >> $GITHUB_ENV
-          echo "DD_PROFILING_LOG_DIR=$tests_output_dir/logs" >> $GITHUB_ENV
-          echo "DD_INTERNAL_PROFILING_OUTPUT_DIR=$tests_output_dir/pprofs" >> $GITHUB_ENV
-          echo "DD_PROFILING_ENABLED=1" >> $GITHUB_ENV
-          echo "DD_PROFILING_EXCEPTION_ENABLED=1" >> $GITHUB_ENV
-          echo "DD_PROFILING_ALLOCATION_ENABLED=1" >> $GITHUB_ENV
-          echo "DD_PROFILING_CONTENTION_ENABLED=1" >> $GITHUB_ENV
-          echo "DD_TRACE_DEBUG=1" >> $GITHUB_ENV
-
-      - name: Run Computer01 for 2min
-        timeout-minutes: 4
-        run: |
-          source profiler/src/ProfilerEngine/DeployResources/DDProf-SetEnv.sh `pwd`/profiler/_build/DDProf-Deploy &&
-          cd profiler/_build/bin/Release-x64/profiler/src/Demos/Samples.Computer01/net6.0 &&
-          # LD_PRELOAD was previously set in the DDProf-SetEnv.sh script, but ASAN does not work with other
-          # library in LD_PRELOAD (segfault)
-          ASAN_OPTIONS="detect_leaks=0" LD_PRELOAD="libasan.so.6" dotnet Samples.Computer01.dll --scenario 1 --timeout 120
+      - uses: ./.github/actions/run-in-docker
+        name: Build profiler with ASAN and run Computer01
+        with:
+          command: "RunProfilerAsanTest --Framework net7.0"
```

Comment:
This duplicates the current tests we're running in the tracer pipeline right? If so I'd be inclined to _not_ change these to use Nuke, so that we continue to test using the "old" approach, in case there's a subtle problem with the Nuke version?